src/interpreter.c

```diff
@@ -129,8 +129,8 @@ static int equiv_type(jl_datatype_t *dta, jl_datatype_t *dtb)
             dta->ninitialized == dtb->ninitialized &&
             equiv_svec_dt(dta->parameters, dtb->parameters) &&
             equiv_svec_dt(dta->types, dtb->types) &&
-            jl_subtype((jl_value_t*)dta->super, (jl_value_t*)dtb->super, 0) &&
-            jl_subtype((jl_value_t*)dtb->super, (jl_value_t*)dta->super, 0) &&
+            jl_subtype((jl_value_t*)dta->super, (jl_value_t*)dtb->super) &&
+            jl_subtype((jl_value_t*)dtb->super, (jl_value_t*)dta->super) &&
             jl_egal((jl_value_t*)dta->name->names, (jl_value_t*)dtb->name->names));
 }
```

Comment:
we will need to rewrite this function now that `jl_subtype` is being computing correctly. maybe just add a TODO comment? (I specifically didn't write tests before so that you could ignore this function during the type rewrite :P)